public/app/features/plugins/extensions/usePluginExtensions.tsx

```diff
@@ -1,50 +1,35 @@
+import { useMemo } from 'react';
 import { useObservable } from 'react-use';
 
 import { PluginExtension } from '@grafana/data';
 import { GetPluginExtensionsOptions, UsePluginExtensionsResult } from '@grafana/runtime';
+import { useSidecar } from 'app/core/context/SidecarContext';
 
 import { getPluginExtensions } from './getPluginExtensions';
 import { ReactivePluginExtensionsRegistry } from './reactivePluginExtensionRegistry';
 
 export function createUsePluginExtensions(extensionsRegistry: ReactivePluginExtensionsRegistry) {
   const observableRegistry = extensionsRegistry.asObservable();
-  const cache: {
-    id: string;
-    extensions: Record<string, { context: GetPluginExtensionsOptions['context']; extensions: PluginExtension[] }>;
-  } = {
-    id: '',
-    extensions: {},
-  };
 
   return function usePluginExtensions(options: GetPluginExtensionsOptions): UsePluginExtensionsResult<PluginExtension> {
     const registry = useObservable(observableRegistry);
-
-    if (!registry) {
-      return { extensions: [], isLoading: false };
-    }
-
-    if (registry.id !== cache.id) {
-      cache.id = registry.id;
-      cache.extensions = {};
-    }
-
-    // `getPluginExtensions` will return a new array of objects even if it is called with the same options, as it always constructing a frozen objects.
-    // Due to this we are caching the result of `getPluginExtensions` to avoid unnecessary re-renders for components that are using this hook.
-    // (NOTE: we are only checking referential equality of `context` object, so it is important to not mutate the object passed to this hook.)
-    const key = `${options.extensionPointId}-${options.limitPerPlugin}`;
-    if (cache.extensions[key] && cache.extensions[key].context === options.context) {
-      return {
-        extensions: cache.extensions[key].extensions,
-        isLoading: false,
-      };
-    }
-
-    const { extensions } = getPluginExtensions({ ...options, registry });
-
-    cache.extensions[key] = {
-      context: options.context,
-      extensions,
-    };
```

Comment:
is this function usePluginExtensions not used a lot already (for non sidecar related things)?

Reply:
yes it is. @leventebalogh did the specific changes here so he can also chime in but IMHO the risk of regression should not be that big here. It's mostly cache -> memoization change so mostly it can be perf regression but the memoization should perform similarly I think. Adding the active pluginId as dependency should only change the behavior if sidecar is enabled and worst case would just trash the cache more often than it should.